snapcraft/internal/lxd.py

```diff
@@ -183,18 +197,69 @@ def _ensure_container(self):
     def _setup_project(self):
         self._ensure_mount(self._project_folder, self._source)
 
+    def _container_login(self):
```

Comment:
I read this thinking that it performed a login. It would be clearer named like _get_container_login.
I would personally call it _get_container_address, but I can see how it could be confusing because of the network address.